transport/tracer_test.go

```diff
@@ -0,0 +1,268 @@
+		ctx,
+		yarpc.NewReqMeta().Procedure("echo"),
+		&echoReqBody{},
+		&resBody,
+	)
+	return err
+}
+
+func (h handler) echoEcho(ctx context.Context) error {
+	var resBody echoResBody
+	_, err := h.client.Call(
+		ctx,
+		yarpc.NewReqMeta().Procedure("echoecho"),
+		&echoReqBody{},
+		&resBody,
+	)
+	return err
+}
+
+func (h handler) createContextWithBaggage(tracer opentracing.Tracer) (context.Context, func()) {
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, 100*time.Millisecond)
+
+	span := tracer.StartSpan("test")
+	// no defer span.Finish()
+	span.SetBaggageItem("weapon", "knife")
+	ctx = opentracing.ContextWithSpan(ctx, span)
+
+	return ctx, cancel
+}
+
+func (h handler) assertBaggage(ctx context.Context) {
+	span := opentracing.SpanFromContext(ctx)
+	weapon := span.BaggageItem("weapon")
+	assert.Equal(h.t, "knife", weapon, "baggage should propagate")
+}
+
+func createHTTPDispatcher(tracer opentracing.Tracer) yarpc.Dispatcher {
+	dispatcher := yarpc.NewDispatcher(yarpc.Config{
+		Name: "yarpc-test",
+		Inbounds: []transport.Inbound{
+			http.NewInbound(":8080"),
+		},
+		Outbounds: transport.Outbounds{
+			"yarpc-test": http.NewOutbound("http://127.0.0.1:8080"),
+		},
+		Tracer: tracer,
+	})
+
+	return dispatcher
+}
+
+func createTChannelDispatcher(tracer opentracing.Tracer, t *testing.T) yarpc.Dispatcher {
+	// Establish the TChannel
+	ch, err := tchannel.NewChannel("yarpc-test", &tchannel.ChannelOptions{
+		Tracer: tracer,
+	})
+	assert.NoError(t, err)
+	hp := "127.0.0.1:4040"
+	ch.ListenAndServe(hp)
+
+	dispatcher := yarpc.NewDispatcher(yarpc.Config{
+		Name: "yarpc-test",
+		Inbounds: []transport.Inbound{
+			ytchannel.NewInbound(ch),
+		},
+		Outbounds: transport.Outbounds{
+			"yarpc-test": ytchannel.NewOutbound(ch, ytchannel.HostPort(hp)),
+		},
+		Tracer: tracer,
+	})
+
+	return dispatcher
+}
+
+func TestHTTPTracer(t *testing.T) {
+	tracer := mocktracer.New()
+	dispatcher := createHTTPDispatcher(tracer)
+
+	client := json.New(dispatcher.Channel("yarpc-test"))
+	handler := handler{client: client, t: t}
+	handler.register(dispatcher)
+
+	dispatcher.Start()
+	defer dispatcher.Stop()
+
+	ctx, cancel := handler.createContextWithBaggage(tracer)
+	defer cancel()
+
+	err := handler.echo(ctx)
+	assert.NoError(t, err)
+
+	AssertDepth1Spans(t, tracer)
+}
+
+func TestTChannelTracer(t *testing.T) {
+	tracer := mocktracer.New()
+	dispatcher := createTChannelDispatcher(tracer, t)
+
+	client := json.New(dispatcher.Channel("yarpc-test"))
+	handler := handler{client: client, t: t}
+	handler.register(dispatcher)
+
+	dispatcher.Start()
+	defer dispatcher.Stop()
+
+	ctx, cancel := handler.createContextWithBaggage(tracer)
+	defer cancel()
+
+	err := handler.echo(ctx)
+	assert.NoError(t, err)
+
+	AssertDepth1Spans(t, tracer)
+}
+
+func AssertDepth1Spans(t *testing.T, tracer *mocktracer.MockTracer) {
+	assert.Equal(t, 2, len(tracer.FinishedSpans()), "generates inbound and outband spans")
+	if len(tracer.FinishedSpans()) != 2 {
```

Comment:
`assert.Equal` returns whether the assertion was successful, so rather than have the check duplicated, you can do:
```go
if !assert.Equal(t, 2, len(...), ...) {
  return
}
```